packaging/os/yum.py

```diff
@@ -603,6 +603,33 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
         cmd = yum_basecmd + ['install'] + pkgs
 
         if module.check_mode:
+            ts = yum.rpm.TransactionSet()
+            ts.setVSFlags(yum.rpm._RPMVSF_NOSIGNATURES)
+            yb = yum_base()
+
+            for package in pkgs:
+                if package.startswith('http'):
+                    tmp_file = tempfile.TemporaryFile()
+
+                    try:
+                        tmp_file, info = fetch_url(module, pkg)
+                        tmp_file.write(rsp.read())
+                    except Exception, e:
+                        module.fail_json(msg="Error downloading %s: %s" % (package, e))
+
+                    tmp_file.seek(0)
+
+                elif os.path.exists(package):
+                    tmp_file = os.open(package, os.O_RDONLY)
+
+                hdr = ts.hdrFromFdno(tmp_file)
+
+                if yb.rpmdb.searchNevra(name=hdr[yum.rpm.RPMTAG_NAME]):
+                    module.exit_json(changed=False)
+
+                if yb.rpmdb.searchNevra(name=package):
+                    module.exit_json(changed=False)
+
```

Comment:
Oh I see what you're trying to do here.  You always create a tmp_file.  then you check both the hdr from ts.hdrFromFdno() and searching by the package name.  I don't think this is the best idea.

1. There's several error conditions that you'd have to catch here to make this logic work.  For instance, ts.hdrFromFdno() run on a non-rpm file (the case if tmp_file is empty because package is neither a url or file) then it will raise an exception.
2. This doesn't handle cleanup of the temporary files that we download.